### DISCUSSION TITLE
REST Query inside JS Transformation | REST API Database

### DISCUSSION THREAD

**Discussion by Kuuchuu** [Budibase/budibase#3875] [Help] [answered]
My Baserow database has a limitation of 200 rows per query. I am hoping to have my javascript transformation loop through several queries so it can combine the data, and return that as the result. I have not had any luck so far trying to GET data from inside my JavaScript transformation. All the functions I have tried so far fail to run. Is there a documented way to perform a new REST query from inside the transformation? Thanks

EDIT: Or is there maybe a way to call another premade REST query from inside the javascript transformation?

**Comment by mjashanks** [comment, marked as answer]
Hi @Kuuchuu 

You won't be able to do this via transformers - they are sandboxed so that you can only do basic scripting.

However, you can use an internal table to iterate through your data:
1. Create a Budibase table called "Baserow Pagination"
2. Add a column called "Page Number" (or whatever data you require to paginate)
3. In your API query, you should add a parameter that accepts the page number (or pagination data)
4. In the `Design` section, add a `Repeater Block`, with the `data` set to `Baserow Pagination`
5. Now, add a `Data Provider` to your API, passing the pagination data from your repeater block in step 4, to your query.

**Reply by tokkisva** [reply]
Hi mjashanks, 

Little bit lost, can you break step 4 down a bit further? 

Thank you!
